autotrash: move to pkgs/by-name and include manpage

MithicSpirit opened the pull request:
## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

Moves autotrash to pkgs/by-name, as well as building and installing its manpage. Also adds myself as the maintainer of this package.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

MithicSpirit committed:
maintainers: add mithicspirit

MithicSpirit committed:
autotrash: include manpage

momeemt commented:
Result of `nixpkgs-review pr 343855` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.autotrash</li>
    <li>python311Packages.autotrash.dist</li>
    <li>python312Packages.autotrash</li>
    <li>python312Packages.autotrash.dist</li>
  </ul>
</details>

momeemt reviewed (approved):
Thank you for your contribution. LGTM

Bot-wxt1221 reviewed (approved):
LGTM. Welcome the nixpkgs!

Sigmanificient commented:
Oh hello Mithic! :D

MithicSpirit committed:
autotrash: move to pkgs/by-name

MithicSpirit commented:
> We should probably move it into `pkgs/by-name` rather than python-modules, since it provides a app, and is separate from python ecosystem, like you said earlier
> 
> Put it under `pkgs/by-name/au/autotrash/package.nix`, and remove the entry line from `pkgs/top-level/python-packages.nix`. You'll also need to prefix some elements with `python3.pkgs` (`buildPythonPackage`, `pytestCheckHook` and python packages passed in `build-system`)

Done. I used `python3Packages` instead of `python3.pkgs` (since this doesn't directly require python3), but lmk if I should change that as well.

Bot-wxt1221 reviewed (approved):
Thanks for your work! Diff LGTM now!

natsukium commented on pkgs/top-level/python-packages.nix:
```diff
@@ -1035,8 +1035,6 @@ self: super: with self; {
 
   autoslot = callPackage ../development/python-modules/autoslot { };
 
-  autotrash = callPackage ../development/python-modules/autotrash { };
```
Could you add an alias to `pkgs/top-level/python-aliases.nix`?

MithicSpirit committed:
autotrash: add to python-aliases

It was moved out of python packages last commit, but this preserves
backwards compatibility.

natsukium reviewed (approved):
LGTM, thanks. Welcome to nixpkgs!🎉
